File: liberacion/solve/cadenas/anagramas.cpp
#include <vector>
#include <string>
#include <iostream>

using namespace std;

vector<int> censoDeLetras(string cadena) {
  vector<int> conteo('z'-'a', 0);
  
  for(int i=0; i<cadena.size(); i++) 
    if(cadena[i] != ' ')
      conteo[cadena[i]-'a']++;

  return conteo;
}

int main() {
  int sw=true; //true es 1
  string S, Q;
  getline(cin, S); getline(cin, Q);

  vector<int> censoS = censoDeLetras(S);
  vector<int> censoQ = censoDeLetras(Q);

  for(int i=0; i<('z'-'a') && sw; i++) 
    if(censoS[i] != censoQ[i])
      sw=false; //false es 0

  if(!sw)
    cout << "NO ";
  cout << "SON ANAGRAMAS" << endl;
}

File: liberacion/solve/cadenas/anagramas.java
import java.util.Scanner;

public class anagramas {

  public static int[] censoDeLetras(String cadena) {
    int[] conteo = new int['z'-'a'];
    
    for(int i=0; i<cadena.length(); i++)
      if(cadena.charAt(i) != ' ')
	conteo[cadena.charAt(i)-'a']++;

    return conteo;
  }
  
  public static void main(String[] args) {
    boolean sw=true;
    String S, Q;
    Scanner lector = new Scanner(System.in);
    S = lector.nextLine();
    Q = lector.nextLine();

    int[] censoS = censoDeLetras(S);
    int[] censoQ = censoDeLetras(Q);
    
    for(int i=0; i<'z'-'a'; i++)
      if(censoS[i] != censoQ[i])
	sw = false;

    if(!sw)
      System.out.print("NO ");
    System.out.println("SON ANAGRAMAS");
  }
}

